packages/blocks/src/api/factory.js

```diff
@@ -530,18 +532,25 @@ export function switchToBlockType( blocks, name ) {
 
 /**
  * Create a block object from the example API.
+ * Can return `undefined` if the block can't be created.
  *
  * @param {string} name
  * @param {Object} example
  *
- * @return {Object} block.
+ * @return {?Object} block object or undefined.
```

Comment:
This syntax is for nullable types.

```suggestion
 * @return {Object|undefined} block object or undefined.
```

https://github.com/WordPress/gutenberg/blob/master/docs/contributors/coding-guidelines.md#nullable-undefined-and-void-types